modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -0,0 +1,580 @@
+            case p: BooleanSchema =>
+              Default(p).extract[Boolean].map(x => Target.pure(new BooleanLiteralExpr(x)))
+            case p: NumberSchema if p.getFormat == "double" =>
+              Default(p).extract[Double].map(x => Target.pure(new DoubleLiteralExpr(x)))
+            case p: NumberSchema if p.getFormat == "float" =>
+              Default(p).extract[Float].map(x => Target.pure(new DoubleLiteralExpr(x)))
+            case p: IntegerSchema if p.getFormat == "int32" =>
+              Default(p).extract[Int].map(x => Target.pure(new IntegerLiteralExpr(x)))
+            case p: IntegerSchema if p.getFormat == "int64" =>
+              Default(p).extract[Long].map(x => Target.pure(new LongLiteralExpr(x)))
+            case p: StringSchema =>
+              Default(p).extract[String].map(x => Target.fromOption(Try(new StringLiteralExpr(x)).toOption, s"Default string literal for '${p.getTitle}' is null"))
+            case _ =>
+              None
+          }): Option[Target[Expression]]).sequence
+
+          readOnlyKey = Option(name).filter(_ => Option(property.getReadOnly).contains(true))
+          emptyToNull = (property match {
+            case d: DateSchema      => ScalaEmptyIsNull(d)
+            case dt: DateTimeSchema => ScalaEmptyIsNull(dt)
+            case s: StringSchema    => ScalaEmptyIsNull(s)
+            case _                  => None
+          }).getOrElse(EmptyIsEmpty)
+
+          tpeClassDep <- meta match {
+            case SwaggerUtil.Resolved(declType, classDep, _) =>
+              Target.pure((declType, classDep))
+            case SwaggerUtil.Deferred(tpeName) =>
+              val tpe = concreteTypes.find(_.clsName == tpeName).map(x => Target.pure(x.tpe)).getOrElse {
+                println(s"Unable to find definition for ${tpeName}, just inlining")
+                safeParseType(tpeName)
+              }
+              tpe.map((_, Option.empty))
+            case SwaggerUtil.DeferredArray(tpeName) =>
+              safeParseType(s"java.util.List<${tpeName}>").map((_, Option.empty))
+            case SwaggerUtil.DeferredMap(tpeName) =>
+              safeParseType(s"java.util.List<${tpeName}>").map((_, Option.empty))
+          }
+          (tpe, classDep) = tpeClassDep
+
+          _declDefaultPair <- Option(isRequired)
+            .filterNot(_ == false)
+            .fold[Target[(Type, Option[Expression])]](
+              (safeParseType(s"java.util.Optional<${tpe}>"), Option(defaultValue.fold[Target[Expression]](safeParseExpression[Expression](s"java.util.Optional.empty()"))(t => safeParseExpression[Expression](s"java.util.Optional.of($t)"))).sequence).mapN((_, _))
+            )(Function.const(Target.pure((tpe, defaultValue))) _)
+          (finalDeclType, finalDefaultValue) = _declDefaultPair
+          term <- safeParseParameter(s"${finalDeclType} ${argName}") // FIXME: How do we deal with default values? .copy(default = finalDefaultValue)
```

Comment:
probably a good idea to `defaultValue: Option[A]`, even in the scala stuff. We do a lot of `case p"$_: $_ = $defval" =>` in the scala code, breaking it out would simplify a lot.